data_steward/gcloud/bq/__init__.py

```diff
@@ -577,3 +577,63 @@ def wait_on_jobs(self,
             sleep(backoff)
             backoff *= 2
         return incomplete_jobs
+
+    def restore_from_time(self,
+                          datasets: typing.List[str],
+                          days_ago: int,
+                          job_config: QueryJobConfig = None) -> datetime:
+        """
+        restore a list of datasets from a specific point in time.
+        tables within each dataset are restored.
+
+        :param datasets: the dataset list
+        :param as_of: The table's contents at specific point in
+        :param job_config: An optional config
+        :return: the time of creation
+        """
+        #* Create, evaluate, and template time information
+        if days_ago > 6:
+            raise ValueError(
+                f'`days_ago` is too far in the past: {days_ago} > 6')
+
+        time_travel_q = JINJA_ENV.from_string("""
+            CREATE TABLE `{{project_id}}.{{dataset_id}}_restore.{{table_id}}_restore`
+            CLONE `{{project_id}}.{{dataset_id}}.{{table_id}}`
+                FOR SYSTEM_TIME AS OF
+                TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL {{days_ago}} DAYS)
+        """)
+
+        #* Labels and Job information
+        job_config = job_config if job_config else QueryJobConfig()
+        job_config.labels.update({'temp': ''})
+
+        for dset in datasets:
+            # for each dataset, get a list of tables and create the recovery dataset
+            # org -> original, rest -> restore
+            org_dset = self.get_dataset(dset)
+            org_tables = list(self.list_tables(org_dset))
+
+            try:
+                rest_dset = self.create_dataset(
+                    f'{org_dset.dataset_id}_restore')
+            except Conflict:
+                raise RuntimeError((
+                    f'The dataset {org_dset.dataset_id}_restore already exists.  '
+                    f'Delete the dataset first to prevent an accidental overwrite.'
+                ))
+            # add the label to include 'temp'
+            rest_dset.labels.update({'temp': ''})
+            self.update_dataset(rest_dset, ['labels'])
+
+            # Generate a list of queries for each table from a template
+            queries = []
+            for table in org_tables:
+                queries.append(
+                    time_travel_q.render(project_id=self.project,
+                                         dataset_id=rest_dset.dataset_id,
+                                         table_id=table.table_id,
+                                         days_ago=days_ago))
+
+            # Execute each query
+            for query in queries:
```

Comment:
@MichaelSchmidt82 This method does what is expected to be done. My only suggestion is let us add all the queries job_ids to a list and wait till all the queries are finished before we move on to next step when this method is called.
```suggestion
            for query in queries:
                job = self.query(query, job_config=job_config)
                job_list.append(job.job_id)
      self.wait_on_jobs(job_list)
      return job_list
```

Reply:
I did not know one could do this.  This is a great way to prevent blocking! thanks.